### PR TITLE
Add placeholders into log messages

### PR DESCRIPTION
**Pull request by hason**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15753 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Although this change is targeted to master, I don't know if Symfony Core members agree with using the short PHP array syntax. This would introduce inconsistency in the codebase.

File: src/Symfony/Component/Routing/Generator/UrlGenerator.php [hunk 1]
```diff
@@ -153,18 +153,18 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
 
         $url = '';
         $optional = true;
+        $message = 'Parameter "{parameter}" for route "{route}" must match "{expected}" ("{given}" given) to generate a corresponding URL.';
```
Comment (Tobion) [inline src/Symfony/Component/Routing/Generator/UrlGenerator.php]: this string is created even if not used on each route. this should not be happening.

**Comment by fabpot** [comment]
We do not use short PHP array syntax in Symfony core.

**Comment by nicolas-grekas** [comment]
:+1:
Status: reviewed

**Comment by fabpot** [comment]
Thank you @hason.

**Comment by lyrixx** [comment]
May be it's too late to discuss about this PR but I'm against it. I really don't see the value added by this (all all stuff related into the PSR + Monolog).

My point of view:

It's very useful for message with a level superior to the level that trigger the cross finger (ie warning / error or more). But for others (debug info) errors it's useless. You have to register to a new processor to do the strtr jobs in order to be able to read the logs. and it does not help anymore with kibana or system like that.

Should I open an issue to discuss about it ?

**Comment by hason** [comment]
@lyrixx Discussion about the benefits is in #15753.

**Comment by lyrixx** [comment]
@hason Yes, I saw it and I replied here. I'm going to copy / paste my previous comment in the issue.

**Pull request merged**